### ISSUE TITLE
Connecting to hosts requiring use of Cisco AnyConnect VPN fails with "no route to host"

### ISSUE THREAD

**Issue by narsimhamchelluri-ds** [microsoft/WSL#8811]
### Version

Microsoft Windows [Version 10.0.22000.918]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.102.1

### Distro Version

Ubuntu 20.04

### Other Software

**Cisco AnyConnect Version:** 4.8.01090
**`wsl.exe` Version:** 10.0.22000.653 (found using [this technique](https://superuser.com/a/1719859))
**Ubuntu WSL package versions (see the note labeled "Ubuntu package names and versions" below for more details):**
- `ubuntu-wsl` version 1.450.2 - `amd64` architecture
- `wslu` version 2.3.6-0ubuntu2~20.04.0 - `all` architecture

### Repro Steps

After booting I open up a WSL terminal window.

Next up I start a PowerShell instance as Admin and run `Get-NetAdapter | Where-Object {$_.InterfaceDescription -Match "Cisco AnyConnect"} | Set-NetIPInterface -InterfaceMetric 6000`. I have ensured that this changes the correct AnyConnect IPv4 InterfaceMetric value to 6000.

Finally I run `Restart-Service LxssManager` from the same PowerShell. This closes my WSL terminal instance, so I open up another one.

Then I run a program from the WSL terminal that attempts to connect to one of the hosts behind the VPN.

### Expected Behavior

I can connect to the hosts behind the VPN.

### Actual Behavior

I get "no route to host" errors.

#### Problem Summary
I'm using Windows 11 with Cisco AnyConnect, as mentioned above. As of a day or so ago my WSL connections to hosts that I access through the VPN have begun failing with the error "no route to host".

#### Is it DNS?
I don't believe this is related to DNS in my case because running `nslookup <hostname>` on either the Windows machine in PowerShell or on the bash prompt in Ubuntu under WSL gives me IPs on the 192.168.x.x range (and the IPs are the same in either case). I believe they are [VPC](https://en.wikipedia.org/wiki/Virtual_private_cloud) hostnames. Moreover, when I run `ipconfig /all` there are no DNS Servers listed under the AnyConnect adapter. I am able to access these VPC hostnames from Windows proper when I am connected to the VPN and I cannot access them when I am not connected to the VPN.

In case it matters, I have taken steps to ensure that my `/etc/resolv.conf` is not auto-generated at WSL startup time. I believe the `generateResolvConf = false` behavior under the `[network]` option in `/etc/wsl.conf` in the version of WSL I use is buggy and I can't recall exactly how I managed to achieve this bit of behavior, but I did - the `resolv.conf` contents don't change between reboots, they don't contain comments about auto-generation, and they are what I have manually set them to, which is nothing interesting: a couple of `nameserver` entries from my WiFi adapter and a `search` entry with that same adapter's Connection-specific DNS Suffix as the value.

#### Are you starting AnyConnect before WSL?
I tried to make sure that AnyConnect does not start when I boot my machine by turning the "Cisco AnyConnect User Interface" setting off in Settings -> Apps -> Startup. I am a little concerned that the "User Interface" means that the AnyConnect program may have started ahead of some UI components but I am not really sure. I am thinking about this because of [this comment](https://github.com/microsoft/WSL/issues/4277#issuecomment-924686361).

#### Ubuntu package names and versions

I ran the command `dpkg -l '*wsl*'` and came up with those package names and versions. I am not positive that this is a comprehensive list of related packages but I did look under the full `dpkg -l` list and tried a few different patterns (`windows`, `subsystem`, etc.). If you have suggestions for more packages to look for I'm happy to try looking for them.

#### Concluding thoughts
Still, after all of this, I am unable to access the VPC hostnames from WSL; I get "no route to host" errors. I reiterate that I can access these hostnames from Windows proper, and I was able to access them from WSL until very recently. In examining my bash history (I keep timestamps) I see that I ran `apt-get upgrade` within the last 24 hours so that may be it.

I have gone through several blog posts and GitHub Gist and Issue comment threads, and I'm at a complete loss. If anyone has any suggestions I would love to hear them. Thank you for your help. I appreciate your time and the effort you have put into making WSL the powerful tool that it is.

**Comment by narsimhamchelluri-ds** [comment]
I am pleased to say I found a temporary workaround.

I adapted the steps described in this video: https://www.youtube.com/watch?v=yR2NsssY7z8

With the WLS Ubuntu running:

1. Disconnect from VPN
2. Open up a PowerShell as Admin
3. Run the following command in the PowerShell to disable the WSL network adapter: `Get-NetAdapter | Where-Object {$_.Name -Match "WSL"} | Disable-NetAdapter -Confirm:$false`
4. Connect to the VPN
5. Run the following command in the PowerShell to enable the WSL network adapter: `Get-NetAdapter | Where-Object {$_.Name -Match "WSL"} | Enable-NetAdapter`

Now I can connect to the hosts on the VPN from WSL again. I _do not_ need to run `Get-NetAdapter | Where-Object {$_.InterfaceDescription -Match "Cisco AnyConnect"} | Set-NetIPInterface -InterfaceMetric 6000` or any other commands.
